weblate/static/editor/tools/search.js

```diff
@@ -0,0 +1,156 @@
+    const $searchForm = $(searchForm);
+    const $searchElment = $searchForm.find(searchElment);
+
+    // Create the preview element
+    const $searchPreview = $('<div id="search-preview"></div>');
+    $searchElment.parent().parent().after($searchPreview);
+
+    let debounceTimeout = null;
+
+    // Update the preview while typing with a debounce of 300ms
+    $searchElment.on("input", () => {
+      $searchPreview.show();
+      const userSearchInput = $searchElment.val();
+      const searchQuery = buildSearchQuery($searchElment);
+
+      // Clear the previous timeout to prevent the previous
+      // request since the user is still typing
+      clearTimeout(debounceTimeout);
+
+      // fetch search results but not too often
+      debounceTimeout = setTimeout(() => {
+        if (userSearchInput) {
+          $.ajax({
+            url: "/api/units/",
+            method: "GET",
+            data: { q: searchQuery },
+            success: (response) => {
+              // Clear previous search results
+              $searchPreview.html("");
+              $("#results-num").remove();
+              const results = response.results;
+              if (!results || results.length === 0) {
+                $searchPreview.text(gettext("No results found"));
+              } else {
+                showResults(results, response.count);
+              }
+            },
+          });
+        }
+      }, 300); // If the user stops typing for 300ms, the search results will be fetched
+    });
+
+    // Show the preview on focus
+    $searchElment.on("focus", () => {
+      if ($searchElment.val() !== "" && $searchPreview.html() !== "") {
+        $searchPreview.show();
+        $("#results-num").show();
+      }
+    });
+
+    // Close the preview on focusout, form submit, form reset, and form clear
+    $searchElment.on("focusout", () => {
+      // Hide after a delay to allow click on a result
+      setTimeout(() => {
+        $searchPreview.hide();
+        $("#results-num").hide();
+      }, 700);
+    });
+    $searchForm.on("submit", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+      $("#results-num").remove();
+    });
+    $searchForm.on("reset", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+      $("#results-num").remove();
+    });
+    $searchForm.on("clear", () => {
+      $searchPreview.html("");
+      $("#results-num").remove();
+      $searchPreview.hide();
+    });
+
+    /**
+     * Handles the search results and displays them in the preview element.
+     * @param {any} results fetched search results
+     * @param {number} count The number of search results
+     * @returns void
+     */
+    function showResults(results, count) {
+      // Show the number of results
+      if (count > 0) {
+        const t = ngettext("%s matching string", "%s matching strings", count);
+        $searchPreview.append(
+          `<h4 id="results-num">${interpolate(t, [count])}</h4>`,
```

Comment:
This could be a link to `/search/?q=...` with the search used in the API.

Reply:
Yeah! okay